src/java/org/apache/cassandra/schema/MemtableParams.java

```diff
@@ -69,6 +69,33 @@ public Memtable.Factory factory()
         return factory;
     }
 
+    /**
+     * Returns a map representation of the memtable configuration for backward compatibility with CC 4.0.
+     * This is used when outputting schema in a format compatible with CC 4.0.
+     *
+     * CC 4.0 accepts both short class names (e.g., 'TrieMemtable') and fully qualified names
+     * (e.g., 'org.apache.cassandra.db.memtable.TrieMemtable'), but using short names is preferred
+     * for consistency and readability.
+     */
+    public Map<String, String> toMapForCC4()
+    {
+        ParameterizedClass definition = CONFIGURATION_DEFINITIONS.get(configurationKey);
+        if (definition != null && definition.class_name != null)
+        {
+            Map<String, String> map = new HashMap<>();
+            String className = definition.class_name;
+            String shortClassName = className.contains(".")
```

Comment:
Converting to the short class name is only possible if it is a class from "org.apache.cassandra.db.memtable.*".  If someone uses a custom class name from a different package it should not be shortened but use the full name.